### PR TITLE
Add code example to doxygen and validate in CI

### PR DESCRIPTION
**Pull request by aggarw13**
Add POSIX example application of using backoffAlgorithm library in doxygen documentation. Also, add a CI check mechanism to validate compilation of code example.

### CODE REVIEW

File: docs/doxygen/pages.dox [hunk 1]
```diff
@@ -61,19 +61,34 @@ every path of execution and achieve 100% branch coverage.
 </p>
 */
 
+/**
+@page backoff_algorithm_example Code Example of backoffAlgorithm API
```
Comment (archigup) [inline docs/doxygen/pages.dox:65]:
```suggestion
@page backoff_algorithm_example Code Example for backoffAlgorithm API
```

File: docs/doxygen/pages.dox [hunk 2]
```diff
@@ -61,19 +61,34 @@ every path of execution and achieve 100% branch coverage.
...
+@brief Example POSIX application of retrying DNS resolution operation with exponential backoff-and-jitter with the backoffAlgorithm library.
```
Comment (archigup) [inline docs/doxygen/pages.dox:66]:
```suggestion
@brief Example POSIX application that retries DNS resolution operation with exponential backoff-and-jitter using the backoffAlgorithm library.
```

File: docs/doxygen/pages.dox [hunk 3]
```diff
@@ -61,19 +61,34 @@ every path of execution and achieve 100% branch coverage.
...
+@include backoff_algorithm_posix.c 
+ */
+
 /**
 @page backoff_algorithm_functions Functions
 @brief Primary functions of the backoffAlgorithm library:<br><br>
 @subpage define_backoffalgorithm_initializeparams <br>
 @subpage define_backoffalgorithm_getnextbackoff <br>
 
+<b>For a code example </b> of using backoffAlgorithm library for retrying operation with exponential back-off and jitter, refer to @ref backoff_algorithm_example.
```
Comment (archigup) [inline docs/doxygen/pages.dox:77]:
```suggestion
<b>For a code example </b> of using backoffAlgorithm library for retrying operations with exponential back-off and jitter, refer to @ref backoff_algorithm_example.
```

**Comment by archigup** [comment]
Is the CMake modification missing?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -35,6 +35,15 @@ jobs:
...
           path: ./build/coverage.info
+  build-code-example:
+  runs-on: ubuntu-latest
+    steps:
+      - name: Clone This Repo
+        uses: actions/checkout@v2
+      - name: Build code example
+        run: |
+          cmake -S test -b Build -DBUILD_CODE_EXAMPLE=ON
+          make -C build code_example_posix -j8
```
Comment (archigup) [inline .github/workflows/ci.yml:46]:
```suggestion
  build-code-example:
    runs-on: ubuntu-latest
      steps:
        - name: Clone This Repo
          uses: actions/checkout@v2
        - name: Build code example
          run: |
            cmake -S test -b Build -DBUILD_CODE_EXAMPLE=ON
            make -C build code_example_posix -j8
```

**Review by archigup — DISMISSED** [review]

File: test/CMakeLists.txt [hunk 5]
```diff
@@ -47,41 +47,53 @@ target_include_directories( coverity_analysis
...
+    )
 
-# Add a target for running coverage on tests.
-add_custom_target( coverage
-    COMMAND ${CMAKE_COMMAND} -DUNITY_DIR=${UNITY_DIR}
-    -P ${MODULE_ROOT_DIR}/tools/unity/coverage.cmake
-    DEPENDS unity backoff_algorithm_utest
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-)
+endif()
```
Comment (paulbartell) [inline test/CMakeLists.txt:99]:
```suggestion
endif()

```

**Review by paulbartell — DISMISSED** [review]

**Review by paulbartell — APPROVED** [review]

**Review by archigup — APPROVED** [review]

**Pull request merged**